experimental/Matrix/matrix.jl

```diff
@@ -0,0 +1,151 @@
+# Initilize GAP function, i.e. GAP reads the file matrix.g
+#
+function __init__()
+    GAP.Globals.Reread(GAP.GapObj(joinpath(Oscar.oscardir, "experimental", "Matrix", "matrix.g")))
+end
+
+################################################################################
+# Computes the JuliaMatrixRep of a MatrixElem in GAP
+#
+#    _wrap_for_gap(m::MatrixElem)
+#
+# Input: m::MatrixElem
+# Output: Returns the JuliaMatrixRep of m in GAP
+#
+# Example
+# ```jldoctest
+# julia> m = matrix(ZZ, [0 1 ; -1 0]);
+# julia> _wrap_for_gap(m)
+# GAP: <matrix object of dimensions 2x2 over Integer Ring>
+# ```
+# """
+function _wrap_for_gap(m::MatrixElem)
+    return GAP.Globals.MakeJuliaMatrixRep(m)
+end
+
+
+################################################################################
+# Construct a GAP group where the elements on the GAP side are wrappers of type
+# JuliaMatrixRep around the Oscar matrices.
+# Moreover, if G is finite then a nice morphism from G into a GAP matrix group G2
+# over a finite field is constructed such that calculations in G can be handled
+# automatically  by transferring them to G2.
+#
+#     MatrixGroup(matrices::Vector{<:MatrixElem{T}}) where T <: Union{fmpz, fmpq, nf_elem}
+#
+# Input: matrices::Vector{<:MatrixElem{T}} where T <: Union{fmpz, fmpq, nf_elem}
+# Output: GAP group generated by the JuliaMatrixReps of matrices
+#
+# Example
+# ```jldoctest
+# julia> m1 = matrix(QQ, [0 1 ; -1 0]);
+# julia> m2 = matrix(QQ, [ -1 0; 0 1]);
+# julia> MatrixGroup([m1,m2])
+# GAP: <group with 2 generators>
+# ```
+# """
+function MatrixGroup(matrices::Vector{<:MatrixElem{T}}) where T <: Union{fmpz, fmpq, nf_elem}
+       @assert !isempty(matrices)
+    
+       # One should probably check whether all matrices are n by n (and invertible
+       # and such ...)
+       
+       n = nrows(matrices[1])
+       isinvertible(x) = applicable(inv, x) && isone(inv(x) * x)
+       isquadraticSameSize(x) = (nrows(x) == n) && (ncols(x) == n)
+       K = base_ring(matrices[1])
+       for mat in matrices
+            if !(K == mat.base_ring)
+                error("Matrices are not from the same base ring.")
+            end
+            if !(isinvertible(mat))
+                error("At least one matrix is not invertible.")
+            end
+            if !(isquadraticSameSize(mat))
+                error("At least one matrix is not quadratic or not the same size.")
+            end
+       end
+       if K isa FlintIntegerRing
+          K = QQ
+       end
+
+       Fq, matrices_Fq, OtoFq = Oscar.good_reduction(matrices, 2)
+
+       # G = Oscar.MatrixGroup(n, Fq, matrices_Fq)
+       #N = order(G)
+       ele = matrices_Fq[1]
+       hom = Oscar._iso_oscar_gap(ele.base_ring)
+        
+       gap_matrices_Fq = GAP.Globals.IdentityMat(length(matrices))
+       for i = 1:length(matrices_Fq)
+            gap_matrices_Fq[i] = AbstractAlgebra.map_entries(hom, matrices_Fq[i])
+       end
```

Comment:
Perhaps like this (note the use of `GAP.Obj` to convert the Julia vector into a GAP list):
```suggestion
       gap_matrices_Fq = GAP.Obj([map_entries(hom, m] for m in matrices_Fq])
```
Note that the `AbstractAlgebra.` prefix should *not* be necessary here